packages/switch/src/Switch.tsx

```diff
@@ -0,0 +1,97 @@
+import { color } from '@sipe-team/tokens';
+import { clsx as cx } from 'clsx';
+import {
+  type CSSProperties,
+  type ComponentProps,
+  type ForwardedRef,
+  forwardRef,
+  useState,
+} from 'react';
+import styles from './Switch.module.css';
+import { type SwitchSize, switchSize } from './constants/size';
+
+export interface SwitchProps extends Omit<ComponentProps<'input'>, 'size'> {
+  defaultChecked?: boolean;
+  checked?: boolean;
+  disabled?: boolean;
+  size?: SwitchSize;
+  onChange?: React.ChangeEventHandler<HTMLInputElement>;
+}
+
+export const Switch = forwardRef(function Switch(
+  {
+    className,
+    defaultChecked,
+    checked: checkedProp,
+    disabled,
+    onChange,
+    size = 'md',
+    ...props
+  }: SwitchProps,
+  ref: ForwardedRef<HTMLInputElement>,
+) {
+  const [internalChecked, setInternalChecked] = useState(
+    defaultChecked ?? false,
+  );
+
+  const isControlled = checkedProp !== undefined;
+  const checked = isControlled ? checkedProp : internalChecked;
+
+  const handleChange: React.ChangeEventHandler<HTMLInputElement> = (event) => {
+    if (!isControlled) {
+      setInternalChecked(event.target.checked);
+    }
+    onChange?.(event);
+  };
+
+  const style = {
+    '--switch-width': `${switchSize[size].width}px`,
+    '--switch-height': `${switchSize[size].height}px`,
+    '--switch-diff': `${switchSize[size].width - switchSize[size].height}px`,
+    '--switch-border-radius': '100px',
+    '--switch-thumb-color': `${color.white}`,
+    '--switch-background-color': `${color.gray300}`,
+    '--switch-checked-background-color': `${color.blue500}`,
+    '--switch-shadow': `0px 2px 4px color-mix(in srgb, ${color.gray900} 10%, transparent), 0px 0px 1px color-mix(in srgb, ${color.gray900} 30%, transparent)`,
+  } as CSSProperties;
+
+  const state = checked ? 'checked' : 'unchecked';
+
+  const dataAttributes = getDataAttributes({
+    disabled,
+  });
+
+  return (
+    <label className={cx(styles['switch-wrapper'], className)} style={style}>
+      <input
+        ref={ref}
+        type="checkbox"
+        className={styles['switch-input']}
+        aria-checked={checked}
+        checked={checked}
```

Comment:
checked가 aria-checked 역할을 대신 해줄 것 같은데 명시적으로 표현해줘야 하는 경우가 있어? 👀

Reply:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-checked

> Note: Where possible use an HTML [<input>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input) element with type="checkbox" and type="radio" as these have built in semantics and do not require ARIA attributes.

이 내용도 차크라를 맹목적으로 신뢰하다가 나와버린 코드..

처음에는 checked 자체로는 aria-checked의 역할과는 다를 것이라고 판단했지만, 
mdn 문서에서는 input checkbox는 시멘틱으로 내장되어 있다고 나와 있음!!

Reply:
`role='switch'` 를 추가해주면서 `aria-checked`가 필수적인지라 다시 부활함!!

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/switch_role
> The ARIA switch role is identical to the [checkbox](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/checkbox_role) role, except instead of being "checked" or "unchecked", it is either "on" or "off". Like the [checkbox](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/checkbox_role) role, the [aria-checked](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-checked) attribute is required.